syntax = "proto3";

package proto;
option go_package = "dp_mock/profile";

message Profile {
    int64 ID = 1; // tg user id, mention [inline mention of a user](tg://user?id=123456789)
    string name = 2;
    int32 age = 3;
    string description = 4;
    string location = 5;
    string pfp_id = 6;
}

message Like {
    int64 user_id = 1;
    int64 liked_user_id = 2;
}

service ProfileService {
    rpc CreateProfile (ProfileRequest) returns (ErrorResponse);
    rpc ReadProfile (IdRequest) returns (Profile);
    rpc UpdateProfile (ProfileRequest) returns (ErrorResponse);

    rpc GetNextProfile (IdRequest) returns (Profile);

    rpc Like (TargetRequest) returns (ErrorResponse);

    rpc GetLikes (Empty) returns (LikesResponse);
}

message ProfileRequest {
    Profile profile = 1;
}

message IdRequest {
    int64 id = 1;
}

message TargetRequest {
    int64 id = 1;
    int64 tgtId = 2;
}

message ErrorResponse {
    string error_message = 1;
}

message LikesResponse {
    repeated Like likes = 1;
}

message Empty {}